stream-chat-android-client/consumer-proguard-rules.pro

```diff
@@ -8,4 +8,15 @@
 # ExtraDataDto can't be minified because we check for extraData using reflection in
 # io.getstream.chat.android.client.parser2.adapters.CustomObjectDtoAdapter. If the name of extraData
 # is changed, we will have problem with serialization.
--keep class * extends io.getstream.chat.android.client.api2.model.dto.ExtraDataDto { *; }
+-keep class * extends io.getstream.chat.android.client.api2.model.dto.ExtraDataDto {
+    public kotlin.collections.Map extraData;
+ }
+
+# Rules necessary for R8 full mode
+-keep class io.getstream.chat.android.client.api2.endpoint.** { *; }
+-keep class io.getstream.chat.android.client.call.RetrofitCall { *; }
+-keep class com.squareup.moshi.JsonReader
+-keep class com.squareup.moshi.JsonAdapter
+-keep class kotlin.reflect.jvm.internal.* { *; }
+-keep class com.xiaomi.mipush.**  { *; }
```

Comment:
Should we keep these rules here or move them to Xiaomi module? 🤔

Reply:
Let's move then

Reply:
Actually, if I move it... it starts breaking o.O